### ISSUE TITLE
axe DevTools flags that ARIA Country and State fields lack accessible name

### ISSUE THREAD

**Issue by paulostp** [woothemes/woocommerce#52540]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Reported internally in 8844477-zen. The issue is observable when the store uses the legacy checkout shortcode, and the "Default customer location" in WooCommerce > Settings > General is "No location by default". This will cause the Country and State fields to be empty when the shopper gets to the Checkout page, leading to the following issue that the axe DevTools Chrome extension reports for both fields.

![Image](https://github.com/user-attachments/assets/137e62af-5cd0-442c-acdb-613c569bb061)

```
Element Location:
#select2-billing_country-container
<span class="select2-selection__rendered" aria-required="true" id="select2-billing_country-container" role="textbox" aria-readonly="true"><span class="select2-selection__placeholder">Select a country / region…</span></span>
```

```
Element Location:
#select2-billing_state-container
<span class="select2-selection__rendered" aria-required="true" id="select2-billing_state-container" role="textbox" aria-readonly="true"><span class="select2-selection__placeholder">Select an option…</span></span>
```

This isn't present if the store uses the Checkout block:

![Image](https://github.com/user-attachments/assets/35de4837-5d52-45d3-85d5-c99d448b008a)

### Expected behavior

No accessibility issues to exist with the checkout fields.

### Actual behavior

axe DevTools reports that the Country and State ARIA input fields don't have an accessible name.

### Steps to reproduce

The issue is observable when the store uses the legacy checkout shortcode, and the "Default customer location" in WooCommerce > Settings > General is "No location by default".

Perform a full scan of the Checkout page using the axe DevTools Chrome extension.

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://profoundly-influential-pony.jurassic.ninja
Site address (URL): https://profoundly-influential-pony.jurassic.ninja
WC Version: 9.3.3
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.1
Log Directory Writable: ✔
WP Version: 6.6.2
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 8.1.30
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 8.10.1
OpenSSL/3.0.14

SUHOSIN Installed: –
MySQL Version: 10.6.19-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 9.3.3
WC Database Prefix: wp_
Total Database Size: 2.89MB
Database Data Size: 1.20MB
Database Index Size: 1.69MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 0.11MB + Index: 0.03MB + Engine InnoDB
wp_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.06MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_orders: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_orders_meta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_addresses: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2
product: 1
revision: 4
shop_order_placehold: 3
wp_global_styles: 1
wp_navigation: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (2) ###

Akismet Anti-spam: Spam Protection: by Automattic - Anti-spam Team – 5.3.3
WooCommerce: by Automattic – 9.3.3

### Dropin Plugins () ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

load.php: by  –

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 88 B

### WC Pages ###

Shop base: #5 - /shop/
Cart: #6 - /cart/ -  Contains the woocommerce/cart block
Checkout: #7 - /checkout/ -  Contains the woocommerce/checkout block
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Four
Version: 1.2
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/error.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/notice.php
/srv/htdocs/wp-content/plugins/woocommerce/templates/block-notices/success.php

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: experimental-blocks
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
blueprint
reactify-classic-payments-settings

Daily Cron: ✔ Next scheduled: 2024-11-05 10:49:49 +00:00
Options: ✔
Notes: 56
Onboarding: -

### Action Scheduler ###

Complete: 7
Oldest: 2024-11-04 10:51:17 +0000
Newest: 2024-11-04 11:45:02 +0000

Failed: 1
Oldest: 2024-11-04 10:51:18 +0000
Newest: 2024-11-04 10:51:18 +0000

Pending: 1
Oldest: 2024-11-05 10:51:17 +0000
Newest: 2024-11-05 10:51:17 +0000

### Status report information ###

Generated at: 2024-11-04 11:45:39 +00:00
`

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by alexflorisca** [comment]
Reproduced this, Country and State should have default ara-labels that say something like "Please select a country/state"

**Comment by AhmarZaidi** [comment]
Hey @paulostp @nerrad, 

I wasn't able to replicate the issue as described on the twenty twenty four theme and latest trunk. 

The `[woocommerce_checkout]` shortcode seems to render the billing country as expected and there are no ARIA related issue in the axe DevTools. Also, setting the "Default customer location" to "No location by default" doesn't seem to unset the Country and State fields. 

![Image](https://github.com/user-attachments/assets/96f50314-1832-47ea-86c5-e836871e7d0a)

https://github.com/user-attachments/assets/b3ef489b-261b-4419-a5e4-4c3e7637d6a7

Could you confirm if there are additional steps to reproduce the issue. 

Thanks!

**Comment by nerrad** [comment]
The original report was on an old version of WooCommerce. There has been a bunch of accessibility work done in the past year including [this PR](https://github.com/woocommerce/woocommerce/pull/57247). Based on that and the report by @AmarZaidi (thanks!) I'm going to go ahead and close this issue.
